### PR TITLE
Move the message restructuring to the end of the block

### PR DESCRIPTION
**Pull request by trink**
- also fail faster by moving the schema tests to the top of the block

### CODE REVIEW

File: heka/sandbox/decoders/extract_telemetry_dimensions.lua [hunk 1]
```diff
@@ -143,21 +143,32 @@ function process_message()
...
-            msg.Payload = json
+
+        -- pull out/verify the data/schema before any restructuring
+        local app = parsed.application
+        if type(app) ~= "table" then
+            return -1, "missing 'application' object"
+        end
+
+        if type(parsed.payload.info) ~= "table" then
+           return -1, "missing info object"
```
Comment (mreid-moz) [inline heka/sandbox/decoders/extract_telemetry_dimensions.lua]: We should be consistent with single-quoting object names (or not) - the messages for missing `application`, `info`, and `creationDate` should look similar.

Comment (mreid-moz) [inline heka/sandbox/decoders/extract_telemetry_dimensions.lua]: I'm not sure we care enough about the `reason` field to make it a decode error if it is missing.

File: heka/sandbox/decoders/extract_telemetry_dimensions.lua [hunk 2]
```diff
@@ -143,21 +143,32 @@ function process_message()
...
-            msg.Payload = json
+
+        -- pull out/verify the data/schema before any restructuring
+        local app = parsed.application
+        if type(app) ~= "table" then
+            return -1, "missing application object"
+        end
+
+        if type(parsed.payload.info) ~= "table" then
+           return -1, "missing info object"
```
Comment (mreid-moz) [inline heka/sandbox/decoders/extract_telemetry_dimensions.lua]: Messages look good now, but I still think we should change this case not to be an error (like `os` below, just use it if it's available).

Comment (trink) [inline heka/sandbox/decoders/extract_telemetry_dimensions.lua]: If it is missing the info OBJECT it should be an error (granted we are only trying to extract reason from it)

Comment (mreid-moz) [inline heka/sandbox/decoders/extract_telemetry_dimensions.lua]: Right, I don't necessarily want to force all future ping types to include an `info` object if they don't have a `reason` field.

**Pull request merged**